packages/server/src/models/UserItemModel.ts

```diff
@@ -138,20 +145,26 @@ export default class UserItemModel extends BaseModel<UserItem> {
 			for (const item of items) {
 				if (!('name' in item) || !('id' in item)) throw new Error('item.id and item.name must be set');
 
-				await super.save({
-					user_id: userId,
-					item_id: item.id,
-				}, options);
-
-				if (this.models().item().shouldRecordChange(item.name)) {
-					await this.models().change().save({
-						item_type: ItemType.UserItem,
-						item_id: item.id,
-						item_name: item.name,
-						type: ChangeType.Create,
-						previous_item: '',
+				try {
+					await super.save({
 						user_id: userId,
-					});
+						item_id: item.id,
+					}, options);
+
+					if (this.models().item().shouldRecordChange(item.name)) {
+						await this.models().change().save({
+							item_type: ItemType.UserItem,
+							item_id: item.id,
+							item_name: item.name,
+							type: ChangeType.Create,
+							previous_item: '',
+							user_id: userId,
+						});
+					}
+				} catch (error) {
+					if (!options.ignoreAlreadyExists || !isUniqueConstraintError(error)) {
```

Comment:
```suggestion
					if (!isUniqueConstraintError(error)) {
```

Some (almost all?) calls to `UserItemModel.add`/`UserItemModel.addMulti` are already wrapped in a `try`/`catch` that ignores unique constraint errors ([example](https://github.com/laurent22/joplin/blob/58a9c229bbf6cbb915887f29a025246aa1c652b8/packages/server/src/models/ShareModel.ts#L194)). It could make sense to remove the option and always ignore unique constraint errors here.